### DISCUSSION TITLE
Table: retrieve all column DataIndex property names

### DISCUSSION THREAD

**Discussion by michaelcsikos** [ant-design-blazor/ant-design-blazor#2951] [Q&A]
Using the `Table.ColumnContext.Columns` I can't see a way to retrieve the `DataIndex` from each column. `Columns` is an `IList<IColumn>` and the `DataIndex` is not in `IColumn` or `ColumnBase`, it is declared in `Column<TData>`.

**Comment by michaelcsikos** [comment]
I have solved this for now by inheriting `Column<TData>` and declaring my own interface:
```cs
public interface IColumnDataIndex
    : IColumn
{
    string DataIndex { get; set; }
}
```

**Reply by michaelcsikos** [reply]
The `ITable` interface should also declare `ColumnContext ColumnContext { get; }`.

**Comment by ElderJames** [comment]
What is your usage scenario?

**Reply by michaelcsikos** [reply]
I have a generic filterable list class which allows multiple properties to be searched simultaneously. I want to retrieve all of the column bound property names from the `Table` to set up the filterable list. The filters support wild cards such as `*` and `?`.

**Reply by ElderJames** [reply]
Oh, that's very nice. you are welcome to creat a PR to add this as a built-in filter.
